capstone/src/main/java/com/google/sps/Objects/Group.java

```diff
@@ -52,15 +52,34 @@ public long getGroupId() {
     return groupId;
   }
 
-  public void addChallenge(Challenge newChallenge) {
+  public void addChallenge(Long newChallenge) {
     this.challenges.add(newChallenge);
   }
 
   public void addMember(String memberId) {
     this.memberIds.add(memberId);
   }
 
-  public void addPost(Post newPost) {
+  public void addPost(Long newPost) {
     this.posts.add(newPost);
   }
+
+  /* Given a Group entity, creates and returns a Group object. */
+  public static Group fromEntity(Entity entity) {
+    ArrayList<String> memberIds = (ArrayList<String>) entity.getProperty("memberIds");
+    ArrayList<Long> challenges = (entity.getProperty("challenges") == null) 
+      ? new ArrayList<Long>() 
+      : (ArrayList<Long>) entity.getProperty("challenges");   
```

Comment:
Could you make this into a helper function which returns the property array<T> you want or an empty array<T> to make this more readable?